cli/azd/cmd/infra_create.go

```diff
@@ -260,6 +264,17 @@ func (ica *infraCreateAction) Run(ctx context.Context, cmd *cobra.Command, args
 	}
 
 	template.CanonicalizeDeploymentOutputs(&res.Result.Properties.Outputs)
+
+	for _, svc := range proj.Services {
+		if svc.Language == "dotnet" {
+			for key, val := range res.Result.Properties.Outputs {
+				if err := tools.NewDotNetCli().SetSecret(ctx, key, fmt.Sprint(val.Value), svc.Path()); err != nil {
```

Comment:
I think you had this code earlier, but it got dropped as you iterated on the PR.

Reply:
Something to think about longer team, we may want to be able to ask a "project" what tools do you require or what tools do you require for these services, as we introduce more abstractions and structure here that might be an interesting thing to explore.  We could union the tools the project system know it needs with the ones the command knows it needs.

@vhvb1989 is also thinking about related issues - no need to do a big overhaul of the system right now, but I do want us to move this check earlier in the process, so we don't fail the overall operation because `dotnet` is missing 90% of the way through completion.